can-validate/map/validate/validate.js

```diff
@@ -325,7 +326,7 @@ proto.__set = function (prop, value, current, success, error) {
 		// If validate opts are set and initing, validate properties only if validateOnInit is true
 		if ((validateOpts && !mapIniting) || (validateOpts && mapIniting && validateOpts.validateOnInit)) {
 			// Validate item
-			allowSet = this._validateOne({key: prop, value: value}, validateOpts);
+			allowSet = this._validateOne({key: prop, value: value}, validateOpts, this.attr());
```

Comment:
Hrm. Bummer. I haven't pulled this down to try it out yet. I'll try it out tomorrow if that's ok.